src/fcc_bridge_node_mavsdk.cpp

```diff
@@ -0,0 +1,274 @@
+    // computer
+    this->mavsdk.emplace(mavsdk::Mavsdk::Configuration(
+        std::get<const u8>(uav_id_map_entry->second),
+        MAV_COMP_ID_ONBOARD_COMPUTER, false));
+
+    // The result of the connection attempt to the MAVLink network will be
+    // stored here
+    mavsdk::ConnectionResult connection_result;
+
+    // Checking if the target UAV is a simulation target
+    if (std::get<const bool>(uav_id_map_entry->second)) {
+        RCLCPP_WARN(this->get_logger(), "Connecting to simulator!");
+        // Trying to connect to the simulated MAVLink network
+        connection_result = this->mavsdk->add_udp_connection(
+            SIMULATOR_IP_ADDRESS, SIMULATOR_PORT_NUMBER,
+            mavsdk::ForwardingOption::ForwardingOff);
+    } else {
+        RCLCPP_DEBUG(this->get_logger(),
+                     "Creating a connection to a real MAVLink network");
+        // Trying to connect to the real MAVLink network
+        connection_result = this->mavsdk->add_serial_connection(
+            UART_DEVICE_PATH, UART_BAUD_RATE, false,
+            mavsdk::ForwardingOption::ForwardingOff);
+    }
+    // Checking if connection succeeded
+    if (connection_result != mavsdk::ConnectionResult::Success) {
+        RCLCPP_FATAL(this->get_logger(),
+                     "Failed to establish MAVSDK connection");
+        RCLCPP_DEBUG(this->get_logger(), "Error code: %d",
+                     static_cast<int>(connection_result));
+        this->internal_state = INTERNAL_STATE::ERROR;
+        return;
+    }
+
+    // Trying to get an autopilot system. There should only be one.
+    const std::optional<std::shared_ptr<mavsdk::System>> potential_system =
+        this->mavsdk->first_autopilot(AUTOPILOT_DISCOVERY_TIMEOUT_S);
+    // Checking if an autopilot was found
+    if (!potential_system.has_value()) {
+        RCLCPP_FATAL(this->get_logger(),
+                     "Could not discover an autopilot within %f seconds",
+                     AUTOPILOT_DISCOVERY_TIMEOUT_S);
+        this->internal_state = INTERNAL_STATE::ERROR;
+        return;
+    }
+    this->mavsdk_system = potential_system.value();
+
+    // Verifying that the system is connected
+    if (!this->mavsdk_system->is_connected()) {
+        RCLCPP_FATAL(this->get_logger(), "Connection to system has failed!");
+        this->internal_state = INTERNAL_STATE::ERROR;
+        return;
+    }
+
+    // Verifying that the system has an autopilot
+    if (!this->mavsdk_system->has_autopilot()) {
+        RCLCPP_FATAL(this->get_logger(),
+                     "The MAVSDK system does not seem to have an autopilot!");
+        this->internal_state = INTERNAL_STATE::ERROR;
+        return;
+    }
+
+    // Setting up mavsdk telemetry
+    this->mavsdk_telemtry.emplace(this->mavsdk_system);
+
+    // Setting up mavsdk action
+    this->mavsdk_action.emplace(this->mavsdk_system);
+
+    // Setting up mavsdk mission
+    this->mavsdk_mission.emplace(this->mavsdk_system);
```

Comment:
Do these throw an exception or how do we know whether they were successfull?

Reply:
And what happens if the mavsdk_system is not valid?

Reply:
Than we error out before that when creating the mavsdk.
I am also checking the mavsdk connection every time I query the FCC.
Unfortunatly I do not know of any way to improve the error checking in this case.